services/rule_engine/src/main/java/agents/Smith.java

```diff
@@ -1,27 +1,77 @@
 package agents;
 
 import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
+import com.mindsmiths.gsheetsAdapter.GSheetsAdapterAPI;
 import com.mindsmiths.gsheetsAdapter.reply.Spreadsheet;
 import com.mindsmiths.ruleEngine.model.Agent;
+import com.mindsmiths.ruleEngine.util.Agents;
 import com.mindsmiths.ruleEngine.util.Log;
 
+
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.ToString;
 
+
 @Data
 @ToString(callSuper = true)
 @AllArgsConstructor
 public class Smith extends Agent {
 
     LocalDateTime lastTableUpdate = null;
+    int nextRow = 0;
+    int odgNext = 0;
```

Comment:
Your naming on these 2 variables is inconsistent. Also, I'm pretty sure you don't need `nextRow` variable

Reply:
Renamed. I think we do because the mail list will not always be the same length as the answer list but that is worked on by Lovro as well.